### PR TITLE
add enrichment fragments

### LINKED ISSUE #102: Build Example for Fragments
### Phase 1 - Content only

- [x] Create an example where a part of a page is loaded from a fragment
- [x] Re-use https://www.hlx.live/developer/block-collection/fragment if possible
- [ ] Please present the result in the next meeting with Hanes

### Phase 2 - Product Enrichment

- [ ] Add product or category metadata to fragments
- [ ] Add an index for fragments including their metadata
- [ ] Create a new block (similar to `fragment`) which can read the product/category identifiers from the URL, looks up matching fragments from the index and displays them
- [ ] Create an example and present the result

### PR DESCRIPTION
**Pull request by hannessolo**
Fix #102 

Test URLs:
- Before: https://main--maidenform--hlxsites.hlx.page/products/a22601
- After: https://enrichment-fragments--maidenform--hlxsites.hlx.page/products/a22601
- Before: https://main--maidenform--hlxsites.hlx.page/products/a22605
- After: https://enrichment-fragments--maidenform--hlxsites.hlx.page/products/a22605
- Before: https://main--maidenform--hlxsites.hlx.page/products/default
- After: https://enrichment-fragments--maidenform--hlxsites.hlx.page/products/default

### CODE REVIEW

File: blocks/product-enrichment/product-enrichment.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { loadFragment } from '../../scripts/scripts.js';
+
+export default async function decorate(block) {
+  const indexRes = await fetch('/fragments/enrichment-fragments.json');
```
Comment (herzog31) [inline blocks/product-enrichment/product-enrichment.js:4]: Please use something like this https://github.com/hlxsites/aldi-nord-recipes/blob/main/scripts/scripts.js#L997-L1035 to optimize the loading of the index file.

**Review by herzog31 — APPROVED** [review]
LGTM

File: blocks/product-enrichment/product-enrichment.js (hunk 1 above)
Comment (hannessolo) [reply blocks/product-enrichment/product-enrichment.js:4]: Does this actually make a performance difference? Does the browser not cache the index?

Comment (herzog31) [reply blocks/product-enrichment/product-enrichment.js:4]: It helps when the index is very long (via pagination) and also prevents multiple blocks from downloading the same index multiple times. The index is only cached after it's loaded, so if two blocks concurrently load the same index, both will result into a request.

**Pull request merged**